internal/controller/nginx/config/policies/snippetspolicy/generator.go

```diff
@@ -0,0 +1,140 @@
+*/
+
+package snippetspolicy
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/nginx/nginx-gateway-fabric/v2/apis/v1alpha1"
+	"github.com/nginx/nginx-gateway-fabric/v2/internal/controller/nginx/config/http"
+	"github.com/nginx/nginx-gateway-fabric/v2/internal/controller/nginx/config/policies"
+)
+
+const (
+	mainTemplate = `
+# SnippetsPolicy %s for Gateway %s main context
+%s
+`
+	httpTemplate = `
+# SnippetsPolicy %s for Gateway %s http context
+%s
+`
+	serverTemplate = `
+# SnippetsPolicy %s for Gateway %s server context in listener %s
+%s
+`
+	locationTemplate = `
+# SnippetsPolicy %s for Gateway %s location context with match key %s
+%s
+`
+)
+
+// Generator generates NGINX configuration snippets for SnippetsPolicies.
+type Generator struct {
+	policies.UnimplementedGenerator
+}
+
+// NewGenerator returns a new instance of Generator.
+func NewGenerator() *Generator {
+	return &Generator{}
+}
+
+// GenerateForMain generates policy configuration for the main block.
+func (g *Generator) GenerateForMain(pols []policies.Policy) policies.GenerateResultFiles {
+	return g.generate(pols, v1alpha1.NginxContextMain, "")
+}
+
+// GenerateForHTTP generates policy configuration for the http block.
+func (g *Generator) GenerateForHTTP(pols []policies.Policy) policies.GenerateResultFiles {
+	return g.generate(pols, v1alpha1.NginxContextHTTP, "")
+}
+
+// GenerateForServer generates policy configuration for the server block.
+func (g *Generator) GenerateForServer(pols []policies.Policy, server http.Server) policies.GenerateResultFiles {
+	return g.generate(pols, v1alpha1.NginxContextHTTPServer, server.Listen)
+}
+
+// GenerateForLocation generates policy configuration for the location block.
+func (g *Generator) GenerateForLocation(pols []policies.Policy, location http.Location) policies.GenerateResultFiles {
+	return g.generate(pols, v1alpha1.NginxContextHTTPServerLocation, location.HTTPMatchKey)
+}
+
+func (g *Generator) generate(
+	pols []policies.Policy,
+	context v1alpha1.NginxContext,
+	id string,
+) policies.GenerateResultFiles {
+	var files policies.GenerateResultFiles
+
+	for _, policy := range pols {
+		sp, ok := policy.(*v1alpha1.SnippetsPolicy)
+		if !ok {
+			continue
+		}
+
+		for _, snippet := range sp.Spec.Snippets {
+			if snippet.Context != context {
+				continue
+			}
+
+			for _, gwRef := range sp.Spec.TargetRefs {
+				// TargetRef info for file path and comment
+				gwNsName := fmt.Sprintf("%s-%s", sp.GetNamespace(), gwRef.Name) // Assuming local target ref implies same namespace
+
+				policyNsName := fmt.Sprintf("%s/%s", sp.GetNamespace(), sp.GetName())
+
+				// Build content and filename
+				var content string
+				var filename string
+
+				switch context {
+				case v1alpha1.NginxContextMain:
+					content = fmt.Sprintf(mainTemplate, policyNsName, gwNsName, snippet.Value)
+					filename = filepath.Join(
+						"includes", "policy", gwNsName,
+						fmt.Sprintf("SnippetsPolicy_main_%s.conf", sp.GetName()),
```

Comment:
namespace needs to be included in the file names. This is to avoid collisions when two snippets policies in different namespaces are used.